File: LeetCode Problems/101 to 200/144. Binary Tree Preorder Traversal/144. Binary Tree Preorder Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:    
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        return [] if not root else   ([root.val]+
                self.preorderTraversal(root.left)+
                self.preorderTraversal(root.right))

File: LeetCode Problems/101 to 200/144. Binary Tree Preorder Traversal/144. Binary Tree Preorder Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
private:
    vector<int> arr;
    void PreorderTraversal(TreeNode *root) {
        if (root == nullptr)
            return;
        arr.push_back(root->val);
        PreorderTraversal(root->left);
        PreorderTraversal(root->right);
    }

public:
    vector<int> preorderTraversal(TreeNode *root) {
        PreorderTraversal(root);
        return arr;
    }
};